### PR TITLE
Add check for empty result in storage memcache

### PR DESCRIPTION
**Pull request by PVince81**
Forward port of https://github.com/owncloud/core/pull/28229 to master

### CODE REVIEW

File: lib/private/Files/Cache/Storage.php [hunk 1]
```diff
@@ -109,7 +109,7 @@ public static function getStorageById($storageId) {
 			self::$localCache = new CappedMemoryCache();
 		}
 		$result = self::$localCache->get($storageId);
-		if ($result === null) {
+		if ($result === null || empty($result) || !isset($result['numeric_id'])) {
```
Comment (jvillafanez) [inline lib/private/Files/Cache/Storage.php]: Any case where the `empty($result)` check is relevant? The `isset` check should be more restrictive anyway.

Comment (PVince81) [inline lib/private/Files/Cache/Storage.php]: `empty($result)` is in case an empty array is returned

Comment (jvillafanez) [inline lib/private/Files/Cache/Storage.php]:
```
php > $a = [];
php > $b = isset($a['key']);
php > var_dump($b);
bool(false)
```
It should be covered with the `isset` check.

Comment (PVince81) [inline lib/private/Files/Cache/Storage.php]: ok, I'll make a master only PR for perfection

Comment (PVince81) [inline lib/private/Files/Cache/Storage.php]: oh PHP... if this was JS that isset would fail

Comment (PVince81) [inline lib/private/Files/Cache/Storage.php]: https://github.com/owncloud/core/pull/28623

**Pull request merged**